src/main/java/com/clouway/push/server/PushServiceImpl.java

```diff
@@ -36,16 +35,13 @@ public void pushEvent(PushEvent event) {
   @Override
   public void pushEvent(PushEvent event, String correlationId) {
 
-    // transforming the eventType
-    if (!Strings.isNullOrEmpty(correlationId)) {
-      event.getAssociatedType().setCorrelationId(correlationId);
-    }
+    String key = event.getKey() + correlationId;
 
     String message = eventSerializer.serialize(event);
```

Comment:
One solution that comes to my mind is to add the correlationId as another argument to the `serialize` method and have it like so:
```java
String message = eventSerializer.serialize(event, correlationId);
```

Another one would be to re-create the PushEvent with the key concatenated with the correlationId.